src/eigenlayer/ZKMRStakeRegistry.sol

```diff
@@ -0,0 +1,210 @@
+        return _getOperatorShares(operator);
+    }
+
+    function getOperatorWeight(address operator)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 weight = _getOperatorShares(operator) / BPS;
+
+        if (weight >= minimumWeight) {
+            return weight;
+        } else {
+            return 0;
+        }
+    }
+
+    function _getOperatorShares(address operator)
+        private
+        view
+        returns (uint256)
+    {
+        uint256 totalShares;
+        StrategyParams[] memory strategyParams = _quorum.strategies;
+        IStrategy[] memory strategies = new IStrategy[](strategyParams.length);
+
+        for (uint256 i; i < strategyParams.length; i++) {
+            strategies[i] = strategyParams[i].strategy;
+        }
+
+        uint256[] memory shares =
+            delegationManager.getOperatorShares(operator, strategies);
+
+        for (uint256 i; i < strategyParams.length; i++) {
+            totalShares += shares[i] * strategyParams[i].multiplier;
+        }
+
+        return totalShares;
+    }
+
+    function _isRegistered(address operator) private view returns (bool) {
+        return operators[operator].x != 0;
+    }
+
+    /// @notice Updates the quorum configuration
+    /// @dev Replaces the current quorum configuration with `newQuorum` if valid.
+    /// Reverts with `InvalidQuorum` if the new quorum configuration is not valid.
+    /// Emits `QuorumUpdated` event with the old and new quorum configurations.
+    /// @param newQuorum The new quorum configuration to set.
+    function _updateQuorumConfig(Quorum memory newQuorum) private {
+        if (!_isValidQuorum(newQuorum)) {
+            revert InvalidQuorum();
+        }
+        Quorum memory oldQuorum = _quorum;
+        delete _quorum;
+        for (uint256 i; i < newQuorum.strategies.length; i++) {
+            _quorum.strategies.push(newQuorum.strategies[i]);
+        }
+        emit QuorumUpdated(oldQuorum, newQuorum);
+    }
+
+    /// @dev Verifies that a specified quorum configuration is valid. A valid quorum has:
+    ///      1. Weights that sum to exactly 10,000 basis points, ensuring proportional representation.
+    ///      2. Unique strategies without duplicates to maintain quorum integrity.
+    /// @param quorum_ The quorum configuration to be validated.
+    /// @return bool True if the quorum configuration is valid, otherwise false.
+    function _isValidQuorum(Quorum memory quorum_)
+        private
+        pure
+        returns (bool)
+    {
+        StrategyParams[] memory strategies = quorum_.strategies;
+        address lastStrategy;
+        address currentStrategy;
+        uint256 totalMultiplier;
+        for (uint256 i; i < strategies.length; i++) {
+            currentStrategy = address(strategies[i].strategy);
+            if (lastStrategy >= currentStrategy) revert NotSorted();
+            lastStrategy = currentStrategy;
+            totalMultiplier += strategies[i].multiplier;
```

Comment:
wait, this is not what is written in the comments. here you're just accumulating the multipliers, which we set to 1, so it'll never be 10,000, no ?
I'm confused both by the logic here and the reason why this function exists.

Reply:
Updated the multipliers to be `1000` each instead of `1` (we have 10 currently)

Originally, I was going to multiply by 1000 after reading from the config file in the deployment script. Now I figured it's best if the config file is explicit (before reading this comment lol)